### ISSUE TITLE
Incredible Memory comsumption for long simulation + ultra slow down

### ISSUE THREAD

**Issue by PyMarc2** [DCC-Lab/PyTissueOptics#66]
The more photons there are to simulate, the slower the simulation gets.
Also, RAM usage can exceed 4GB on my end. 

```
from pytissueoptics import *


materialLayer1 = ScatteringMaterial(mu_s=1.0, mu_a=0.5, g=0.7, n=1.3)
materialLayer2 = ScatteringMaterial(mu_s=5.0, mu_a=0.8, g=0.8, n=1.4)
materialLayer3 = ScatteringMaterial(mu_s=50.0, mu_a=2.5, g=0.9, n=1.5)

layer1 = Cuboid(a=50, b=50, c=1, position=Vector(0, 0, 0), material=materialLayer1, label="Layer 1")
layer2 = Cuboid(a=50, b=50, c=1, position=Vector(0, 0, 0), material=materialLayer2, label="Layer 2")
layer3 = Cuboid(a=50, b=50, c=1, position=Vector(0, 0, 0), material=materialLayer3, label="Layer 3")
stack1 = layer1.stack(layer2, "back")
stackedTissue = stack1.stack(layer3, "back")

layerTissueScene = RayScatteringScene([stackedTissue])

logger = Logger()
source = PencilSource(position=Vector(0, 0, -5), direction=Vector(0, 0, 1), N=1000)
source.propagate(layerTissueScene, logger)

stats = Stats(logger, source, layerTissueScene)
stats.report("Layer 1")
stats.report("Layer 2")
stats.report("Layer 3")

```

This particular simulation starts very fast, but slows down to almost a halt at the end ( 3-4 photons/s)

**Comment by PyMarc2** [comment]
This is due to the np.concatenate which is O(N) time complexity. So the more data the slower it gets. However, we still need numpy array for GPU logging since converting large numpy array to list makes memory errors.

We are working on a solution

**Comment by PyMarc2** [comment]
closed with #69
